src/sagas/profile.test.js

```diff
@@ -0,0 +1,324 @@
+import MD5 from 'md5.js';
+import request from 'utils/request';
+import { actions as appActions, types as appTypes } from 'reducers/application';
+import { actions } from 'reducers/profile';
+import { actions as toastActions } from 'reducers/toast';
+import saga, { getCurrentUser, watchers, workers } from './profile';
+import { getUserMap, getUserProfiles } from 'selectors/profile';
+import { getUser } from 'selectors/application';
+
+describe('Profile sagas', () => {
+  const currentUser = {
+    currentUser: true,
+    id: 3,
+    email: 'd@d.com'
+  };
+
+  const user = {
+    name: 'mixn',
+    currentUser: true
+  };
+
+  it('handles success in getCurrentUser', () => {
+    const gen = getCurrentUser();
+
+    let result = gen.next();
+
+    expect(result.value).toEqual(select(getUser));
+
+    result = gen.next();
+
+    expect(result.value).toEqual(put(appActions.requestCurrentUser()));
+
+    result = gen.next(
+      take([
+        appTypes.REQUEST_CURRENT_USER_SUCCESS,
+        appTypes.REQUEST_CURRENT_USER_FAILURE
+      ])
+    );
+
+    expect(result.value).toEqual(
+      take([
+        appTypes.REQUEST_CURRENT_USER_SUCCESS,
+        appTypes.REQUEST_CURRENT_USER_FAILURE
+      ])
+    );
+  });
+
+  const data = {
+    userId: 3,
+    name: 'mixn',
+    bio: 'testing',
+    location: 'testing, pl',
+    url: 'test.com',
+    User: { emailAddress: 'd@d.com' }
+  };
+
+  const profile = {
+    name: 'mixn',
+    bio: 'testing',
+    location: 'testing, pl',
+    url: 'test.com',
+    gravatar: new MD5().update('d@d.com').digest('hex')
+  };
+
+  /* const profiles = [
+    {
+      userId: 1
+    },
+    {
+      userId: 2
+    },
+    {
+      userId: 3
+    }
+  ];*/
+
+  const usernameEndpoint = {
+    url: `/user/mixn`,
+    method: 'GET'
+  };
+
+  /* const profileEndpoint = {
+    url: `/user/3/profile`,
+    method: 'GET'
+  };*/
+
+  const map = [
+    { user1: { userId: 1 } },
+    { user2: { userId: 2 } },
+    { mixn: { userId: 3 } }
+  ];
+
+  it('handles success in requestProfileWorker', () => {
+    const gen = workers.requestProfileWorker({ user });
+
+    let result = gen.next(user);
+
+    expect(result.value).toEqual(call(getCurrentUser));
+
+    result = gen.next(user);
+
+    expect(result.value).toEqual(select(getUserProfiles));
+
+    result = gen.next(map);
+
+    expect(result.value).toEqual(select(getUserMap));
+
+    result = gen.next(map);
+
+    expect(result.value).toEqual(
+      call(request.execute, { endpoint: usernameEndpoint })
+    );
+
+    result = gen.next({
+      success: true,
+      response: {
+        data
+      }
+    });
+
+    expect(result.value).toEqual(
+      put(
+        actions.requestProfileSuccess({
+          request: profile,
+          cache: map,
+          currentUser: true
+        })
+      )
+    );
+
+    result = gen.next();
+
+    expect(result.value).toEqual(put(actions.mapUser(map)));
+
+    result = gen.next();
```

Comment:
Why is nothing being passed in here?

Reply:
Refactored, I'm beginning to dislike the flow of this saga in general, so I'll likely change the way this works with the next phase of profiles.